plugin-server/src/main/ingestion-queues/session-recording/blob-ingester/session-manager.ts

```diff
@@ -384,11 +389,16 @@ export class SessionManager {
     }
 
     private async destroyBuffer(buffer: SessionBuffer): Promise<void> {
-        buffer.fileStream.destroy()
-        await unlink(buffer.file).catch((error) => {
-            status.error('🧨', 'blob_ingester_session_manager failed to unlink buffer file', {
-                ...this.logContext(),
-                error,
+        await new Promise<void>((resolve) => {
+            buffer.fileStream.close(async () => {
+                try {
+                    await stat(buffer.file)
+                    await unlink(buffer.file)
+                } catch (error) {
+                    // Indicates the file was already deleted (i.e. if there was never any data in the buffer)
```

Comment:
_really_ we should only swallow some exceptions here, but you'd expect any new problem that meant we couldn't delete would show up some other way so 🤷